src/main/java/com/playmonumenta/networkchat/PlayerStateManager.java

```diff
@@ -214,10 +215,21 @@ public static void unregisterChannel(UUID channelId) {
 		}
 	}
 
+	@EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
+	public void playerLoginEvent(PlayerLoginEvent event) {
+		Player player = event.getPlayer();
+		String playerName = player.getName();
+		if (NetworkChatPlugin.globalBadWordFilter().hasBadWord(player, Component.text(playerName))) {
+			event.kickMessage(Component.text("Please change your player name", NamedTextColor.RED));
+			event.setResult(PlayerLoginEvent.Result.KICK_OTHER);
```

Comment:
You could use the `event.disallow` method to combine this
[disallow](https://jd.papermc.io/paper/1.19/org/bukkit/event/player/PlayerLoginEvent.html#disallow(org.bukkit.event.player.PlayerLoginEvent.Result,net.kyori.adventure.text.Component))

Reply:
So *that's* how it's supposed to be done - I didn't open the JavaDoc, and scrolling through the methods in my IDE was a bit tedious.